e2e/nomostest/prometheus_metrics.go

```diff
@@ -344,56 +344,54 @@ func metricResourceFightsHasValueAtLeast(nt *NT, syncLabels prometheusmodel.Labe
 		labels := prometheusmodel.LabelSet{
 			prometheusmodel.LabelName(ocmetrics.KeyComponent.Name()): prometheusmodel.LabelValue(ocmetrics.OtelCollectorName),
 		}.Merge(syncLabels)
-		// ResourceFightsView counts the total number of ResourceFights, so we don't need to aggregate
-		query := fmt.Sprintf("%s%s", metricName, labels)
 		if value == 0 {
-			// Tolerate missing metrics when expecting a zero value.
-			// Don't allow any value other than zero.
-			return metricExistsWithValueOrDoesNotExist(ctx, nt, v1api, query, float64(value))
+			query := fmt.Sprintf("rate(%s%s[1m])", metricName, labels)
+			return metricExistsWithValue(ctx, nt, v1api, query, 0)
 		}
+		query := fmt.Sprintf("%s%s", metricName, labels)
 		return metricExistsWithValueAtLeast(ctx, nt, v1api, query, float64(value))
 	}
 }
 
 // metricResourceConflictsHasValueAtLeast returns a MetricsPredicate that
-// validates that ResourceConflicts has at least the expected value.
-// If the expected value is zero, the metric must be zero or not found.
+// validates whether ResourceConflicts has been recorded recently
+// If the expected value is zero, the rate at which ResourceConflicts occur is also zero.
 func metricResourceConflictsHasValueAtLeast(nt *NT, syncLabels prometheusmodel.LabelSet, commitHash string, value int) MetricsPredicate {
 	return func(ctx context.Context, v1api prometheusv1.API) error {
 		metricName := ocmetrics.ResourceConflictsView.Name
 		metricName = fmt.Sprintf("%s%s", prometheusConfigSyncMetricPrefix, metricName)
+		if value == 0 {
+			labels := prometheusmodel.LabelSet{
+				prometheusmodel.LabelName(ocmetrics.KeyComponent.Name()): prometheusmodel.LabelValue(ocmetrics.OtelCollectorName),
+				prometheusmodel.LabelName(ocmetrics.KeyCommit.Name()):    ocmetrics.Initialization,
+			}.Merge(syncLabels)
+			query := fmt.Sprintf("rate(%s%s[30s])", metricName, labels)
```

Comment:
Why use 30s here and 1m for the other call? Seems like they should be the same, and maybe configurable by serializing a time.Durration argument.

Reply:
I was manually tuning each so forgot to align the interval, we could pass an argument duration but then need to convert it to the verbal time with unit.. Or we just use seconds to make things easier.

Reply:
time.Duration.String() should give you want you need. Prometheus is written in Go anyway.

Reply:
This part has been moved to https://github.com/GoogleContainerTools/kpt-config-sync/pull/1237, will be making change accordingly.